src/Reports/SiteSummary.php

```diff
@@ -84,6 +88,30 @@ public function getReportField()
         return $grid;
     }
 
+    public function getCMSFields()
+    {
+        $fields = parent::getCMSFields();
+        $alerts = $this->getAlerts();
+        if ($alerts) {
+            $summaryInfo = '<div class="site-summary message warning">' . implode("\n", $alerts) . '</div>';
+            $fields->unshift(LiteralField::create('AlertSummary', $summaryInfo));
+        }
```

Comment:
If you were to put this into a `beforeUpdateCMSFields` callback then extensions could remove or adjust it if they wanted to. They won't be able to at the moment

Reply:
>Is that a thing in 3?

Yep:

```php
public function getCMSFields()
{
    $this->beforeUpdateCMSFields(function (FieldList $fields) {
        // ...  modify field list here
    });
    return parent::getCMSFields();
}
```

Reply:
Yeah nah, this is a `SS_Report` though, not a `DataObject` (the only place the function is defined) :>

Me test give us an error: 
```
Exception: Object->__call(): the method 'beforeupdatecmsfields' does not exist on 'SiteSummary', or the method is not public.
```

Hooray for tests! :D

Reply:
Ah yep, you can still use beforeExtending callbacks if you want here but it’s ok :-D